src/jolt/subtable/sra_sign.rs

```diff
@@ -0,0 +1,96 @@
+use ark_ff::PrimeField;
+use ark_std::log2;
+use std::marker::PhantomData;
+use std::sync::Arc;
+
+use super::LassoSubtable;
+use crate::utils::math::Math;
+use crate::utils::split_bits;
+
+#[derive(Default)]
+pub struct SraSignSubtable<F: PrimeField> {
+  _field: PhantomData<F>,
+}
+
+impl<F: PrimeField> SraSignSubtable<F> {
+  pub fn new() -> Self {
+    Self {
+      _field: PhantomData,
+    }
+  }
+}
+
+impl<F: PrimeField> LassoSubtable<F> for SraSignSubtable<F> {
+  fn materialize(&self, M: usize) -> Vec<F> {
+    let mut entries: Vec<F> = Vec::with_capacity(M);
+
+    let operand_chunk_width: usize = (log2(M) / 2) as usize;
+
+    // find position of sign bit in the chunk 
+    let sign_bit_index = 63 % operand_chunk_width; 
+    let sign_bit_mask: usize = 1 << (sign_bit_index);
+
+    for idx in 0..M {
+      let (x, y) = split_bits(idx, operand_chunk_width);
+
+      let x_sign = F::from(((x & sign_bit_mask) >> sign_bit_index) as u64);
```

Comment:
slightly cleaner imo
```suggestion
      let x_sign = F::from(((x >> sign_bit_index) & 1) as u64);
```